deepdrivemd/misc/aggregator_reader.py

```diff
@@ -0,0 +1,193 @@
+import adios2
+import numpy as np
+from deepdrivemd.utils import t1Dto2D
+
+
+class ADIOS_READER:
+    def __init__(self, fn, config, stream_name):
+        self.adios = adios2.ADIOS(str(config), True)
+        self.io = self.adios.DeclareIO(stream_name)
+        self.stream = self.io.Open(fn, adios2.Mode.Read)
+
+    def __del__(self):
+        self.stream.Close()
+        self.io.RemoveAllVariables()
+        self.adios.RemoveAllIOs()
+
+    def next_all(self, N):
+        CMs = []
+        POSITIONs = []
+        MD5s = []
+        STEPs = []
+        RMSDs = []
+        VELOCITYs = []
+        MD5s = []
+        for i in range(N):
+            status = self.stream.BeginStep(adios2.StepMode.Read, 0.0)
+            if status != adios2.StepStatus.OK:
+                break
+
+            step = np.zeros(1, dtype=np.int32)
```

Comment:
Each variable inquire is code duplication. We can import the functions get_scalar and get_array from the previous comment. This will really help to start building out our I/O abstractions and APIs.

Reply:
Same thing: between BeginStep() and EndStep() things are running asynchronously and the data transfer is guaranteed to finish only after EndStep(). Inserting functions in between is likely asking for trouble.

Reply:
See new design